storage: TestEndTransactionWithErrors is flaky

Took about 4000 runs to reproduce this locally, but I also saw it in circleCI:
```
I1110 14:28:27.285909 16458 multiraft/multiraft.go:929  node 1 campaigning because initial confstate is [1]
I1110 14:28:27.287996 16458 storage/replica_command.go:1155  range 1: new leader lease replica {1 1 1} 1970-01-01 00:00:00 +0000 UTC +1.000s
I1110 14:28:27.299330 16458 storage/engine/rocksdb.go:132  closing in-memory rocksdb instance
--- FAIL: TestEndTransactionWithErrors (0.03s)
	replica_test.go:1938: expected error:
		txn "test" id=fa81a303 key="d" rw=false pri=33.54399467 iso=SERIALIZABLE stat=PENDING epo=0 ts=0.000000000,12 orig=0.000000001,1 max=0.000000001,1: timestamp regression: 0.000000001,1
		to match:
		txn "test" id=.*: timestamp regression: 0.000000001,0
FAIL


ERROR: exit status 1
```

--- a/storage/replica_test.go
+++ b/storage/replica_test.go
@@ -1908,7 +1908,7 @@ func TestEndTransactionWithErrors(t *testing.T) {
 		{roachpb.Key("a"), roachpb.COMMITTED, txn.Epoch, txn.Timestamp, "txn \"test\" id=.*: already committed"},
 		{roachpb.Key("b"), roachpb.ABORTED, txn.Epoch, txn.Timestamp, "txn aborted \"test\" id=.*"},
 		{roachpb.Key("c"), roachpb.PENDING, txn.Epoch + 1, txn.Timestamp, "txn \"test\" id=.*: epoch regression: 0"},
-		{roachpb.Key("d"), roachpb.PENDING, txn.Epoch, regressTS, "txn \"test\" id=.*: timestamp regression: 0.000000001,0"},
+		{roachpb.Key("d"), roachpb.PENDING, txn.Epoch, regressTS, `txn "test" id=.*: timestamp regression: 0.000000001,\d+`},
 	}
 	for _, test := range testCases {
 		// Establish existing txn state by writing directly to range engine.

